pkg/functions/deploy.go

```diff
@@ -73,10 +59,62 @@ func RunDeploy(w io.Writer, sdk *SdkStatus) error {
 			if err != nil {
 				return err
 			}
+		} else if step.Build.Builder != "" {
+			if step.Build.Destination == "" {
+				return fmt.Errorf("Please provide a destination for the build image.")
+			}
+			if step.Build.MainPath == "" {
+				return fmt.Errorf("Please provide the relative path to the main file.")
+			}
+			// TODO(shefaliv): remove the GOOGLE_BUILDABLE dependency
+			packCommand := fmt.Sprintf("pack build %s --builder %s --env GOOGLE_BUILDABLE=%s --publish", step.Build.Destination, step.Build.Builder, step.Build.MainPath)
+			if err := runCommand(packCommand); err != nil {
+				return err
+			}
+
+			data := make(map[string]interface{})
+			data["MetadataName"] = deployPlan.ObjectMeta.Name
+			data["Image"] = step.Build.Destination
+			box := packr.NewBox("./template")
```

Comment:
the way the build is done needs to be changed for this to be effective - i think this is fine for now, but we'll want to address packaging static assets correctly at some point.